sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/firestore/QueryUtilsTest.java

```diff
@@ -0,0 +1,185 @@
+                                    .addFilters(
+                                        Filter.newBuilder()
+                                            .setFieldFilter(
+                                                FieldFilter.newBuilder()
+                                                    .setField(
+                                                        FieldReference.newBuilder()
+                                                            .setFieldPath("z.a.a"))
+                                                    .setOp(FieldFilter.Operator.NOT_EQUAL)
+                                                    .setValue(
+                                                        Value.newBuilder().setStringValue(""))))
+                                    .addFilters(
+                                        Filter.newBuilder()
+                                            .setFieldFilter(
+                                                FieldFilter.newBuilder()
+                                                    .setField(
+                                                        FieldReference.newBuilder()
+                                                            .setFieldPath("b"))
+                                                    .setOp(FieldFilter.Operator.GREATER_THAN)
+                                                    .setValue(
+                                                        Value.newBuilder().setStringValue(""))))
+                                    .setOp(CompositeFilter.Operator.AND)))
+                    .addFilters(
+                        Filter.newBuilder()
+                            .setFieldFilter(
+                                FieldFilter.newBuilder()
+                                    .setField(FieldReference.newBuilder().setFieldPath("c"))
+                                    .setOp(FieldFilter.Operator.EQUAL)
+                                    .setValue(Value.newBuilder().setStringValue(""))))
+                    .addFilters(
+                        Filter.newBuilder()
+                            .setUnaryFilter(
+                                UnaryFilter.newBuilder()
+                                    .setField(FieldReference.newBuilder().setFieldPath("z"))
+                                    .setOp(UnaryFilter.Operator.IS_NOT_NAN)))
+                    .setOp(CompositeFilter.Operator.AND)
+                    .build());
+    testQuery =
+        StructuredQuery.newBuilder()
+            .addFrom(
+                CollectionSelector.newBuilder()
+                    .setAllDescendants(false)
+                    .setCollectionId("collection"))
+            .setWhere(filter)
+            .addOrderBy(
+                Order.newBuilder()
+                    .setField(FieldReference.newBuilder().setFieldPath("b"))
+                    .setDirection(Direction.DESCENDING))
+            .build();
+  }
+
+  @Test
+  public void getImplicitOrderBy_success() {
```

Comment:
This could use some extra test case, for example:
- multiple equivalent field paths on inequalities (\`foo\` > 1 AND foo < 2)
- inequalities on `__name__` + other field
- orderby partially matching some inequalities / `__name__`
- order by matching but different escaping: WHERE \`foo\` > 1 ORDER BY foo

Reply:
\#3 was already in there, I've added another test case for \#2 and tweaked the main one to cover \#1 and \#4.